clientgui/BOINCTaskBar.cpp

```diff
@@ -412,13 +412,19 @@ wxMenu *CTaskBarIcon::CreatePopupMenu() {
 // so we don't need additional Snooze and Disconnected icons for branding.
 bool CTaskBarIcon::SetIcon(const wxIcon& icon, const wxString& ) {
     wxImage macIcon;
-    bool result;
     int err = noErr;
     int w, h, x, y;
 
     if (m_iconType != wxTBI_DOCK) {
-        result = wxGetApp().GetMacDockIcon()->SetIcon(icon);
-        return (result && wxTaskBarIcon::SetIcon(icon));
+        bool displayMenuBarExtra = wxGetApp().GetBOINCMGRDisplayMenuBarExtra();
+        if (displayMenuBarExtra) {
+            bool result;
```

Comment:
No need to create this local variable, `wxGetApp().GetMacDockIcon()->SetIcon(icon)` could be moved into return statement.

Reply:
Done: ae543d9c1d